### ISSUE TITLE
Issue with exportsd.py for optimizers when no state has been calculated

### ISSUE THREAD

**Issue by shaltielshmid** [dotnet/TorchSharp#1153]
In PyTorch, if a given parameter hasn't had any steps taken after a gradient has been calculated for that parameter, then the `state_dict()` of some optimizers will not have a value in the state for it, it will just return an empty dictionary. In the code this feature is commented as "Lazy state initialization". 
The `exportsd.py` code assumes that there is a state for each of the parameters, so it tries to look up the different values in the state that is empty. 

Code to reproduce:
```
import torch
import exportsd
lin = torch.nn.Linear(10, 10)
opt = torch.optim.Adam(lin.parameters())

f = open("adam_state.dat", "wb")
exportsd.save_adam(opt, f)
f.close()
```
